diff --git a/tar.c b/tar.c
index 10b2e30..8fd89a0 100644
--- a/tar.c
+++ b/tar.c
@@ -568,7 +568,7 @@ main(int argc, char *argv[])
 	case 't':
 	case 'x':
 		tarfd = 0;
-		if (file) {
+		if (file && 0 != strcmp("-",file)) {
 			tarfd = open(file, O_RDONLY);
 			if (tarfd < 0)
 				eprintf("open %s:", file);
diff --git a/touch.c b/touch.c
index b957fa5..4217dfb 100644
--- a/touch.c
+++ b/touch.c
@@ -146,6 +146,8 @@ main(int argc, char *argv[])
 	case 'T':
 		times[0].tv_sec = estrtonum(EARGF(usage()), 0, LLONG_MAX);
 		break;
+	case 'f':
+	    break;
 	default:
 		usage();
 	} ARGEND
